dockstore-integration-testing/src/test/java/io/dockstore/webservice/metrics/MetricsDataS3ClientIT.java

```diff
@@ -56,137 +74,181 @@
 @LocalstackDockerProperties(imageTag = LOCALSTACK_IMAGE_TAG, services = { ServiceName.S3 }, environmentVariableProvider = MetricsDataS3ClientIT.LocalStackEnvironmentVariables.class)
-public class MetricsDataS3ClientIT {
+public class MetricsDataS3ClientIT extends BaseIT {
     public static final String LOCALSTACK_IMAGE_TAG = "1.3.1";
-    private static final String BUCKET_NAME = "dockstore.metrics.data";
+    private static final Gson GSON = new Gson();
+    private static String bucketName;
     private static S3Client s3Client;
+    private static MetricsDataS3Client metricsDataClient;
 
     @BeforeAll
     public static void setup() throws URISyntaxException {
+        bucketName = SUPPORT.getConfiguration().getMetricsConfig().getS3BucketName();
         s3Client = TestUtils.getClientS3V2(); // Use localstack S3Client
+        metricsDataClient = new MetricsDataS3Client(bucketName, s3Client);
         // Create a bucket to be used for tests
-        CreateBucketRequest request = CreateBucketRequest.builder().bucket(BUCKET_NAME).build();
+        CreateBucketRequest request = CreateBucketRequest.builder().bucket(bucketName).build();
         s3Client.createBucket(request);
         deleteBucketContents(); // This is here just in case a test was stopped before tearDown could clean up the bucket
     }
 
+    @BeforeEach
+    @Override
+    public void resetDBBetweenTests() throws Exception {
+        CommonTestUtilities.cleanStatePrivate2(SUPPORT, false, testingPostgres);
+        // Set the MetricsDataS3Client to use the localstack S3 client for testing
+        ToolsApiExtendedServiceImpl.setMetricsDataS3Client(new MetricsDataS3Client(SUPPORT.getConfiguration().getMetricsConfig().getS3BucketName(), s3Client));
+    }
+
     @AfterEach
     public void tearDown() {
         // Delete all objects from the S3 bucket after each test
         deleteBucketContents();
     }
 
     private static void deleteBucketContents() {
-        ListObjectsV2Request request = ListObjectsV2Request.builder().bucket(BUCKET_NAME).build();
+        ListObjectsV2Request request = ListObjectsV2Request.builder().bucket(bucketName).build();
         ListObjectsV2Response response = s3Client.listObjectsV2(request);
         List<S3Object> contents = response.contents();
         contents.forEach(s3Object -> {
-            DeleteObjectRequest deleteObjectRequest = DeleteObjectRequest.builder().bucket(BUCKET_NAME).key(s3Object.key()).build();
+            DeleteObjectRequest deleteObjectRequest = DeleteObjectRequest.builder().bucket(bucketName).key(s3Object.key()).build();
             s3Client.deleteObject(deleteObjectRequest);
         });
     }
 
     /**
-     * This test is a prototype for how metric data would be stored in S3.
+     * Test submitting metrics data using the Extended GA4GH endpoint.
      *
      * At the end of this test, the S3 folder structure should look like the following:
-     * dockstore.metrics.data
+     * local-dockstore-metrics-data
```

Comment:
If I understand correctly, should also document what information is saved where using metadata tags